modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java

```diff
@@ -1059,6 +1059,88 @@ public void testAddObjectEntryWithFormulaObjectField() throws Exception {
 	@Test
 	public void testAddObjectEntryWithObjectValidationRule() throws Exception {
 
+		// Composite Key field values must be unique
+
+		ObjectField objectField1 = _objectFieldLocalService.fetchObjectField(
+			_objectDefinition.getObjectDefinitionId(), "emailAddressRequired");
+
+		ObjectField objectField2 = _objectFieldLocalService.fetchObjectField(
+			_objectDefinition.getObjectDefinitionId(),
+			"listTypeEntryKeyRequired");
+
+		ObjectField objectField3 = _objectFieldLocalService.fetchObjectField(
+			_objectDefinition.getObjectDefinitionId(), "numberOfBooksWritten");
+
+		ObjectValidationRule objectValidationRule6 = _addObjectValidationRule(
+			ObjectValidationRuleConstants.ENGINE_TYPE_COMPOSITE_KEY,
+			LocalizedMapUtil.getLocalizedMap(
+				"Composite Key field values must be unique"),
+			ObjectValidationRuleConstants.OUTPUT_TYPE_FULL_VALIDATION, "",
+			Arrays.asList(
+				new ObjectValidationRuleSettingBuilder(
+				).name(
+					ObjectValidationRuleSettingConstants.
+						NAME_COMPOSITE_KEY_OBJECT_FIELD_ID
+				).value(
+					String.valueOf(objectField1.getObjectFieldId())
+				).build(),
+				new ObjectValidationRuleSettingBuilder(
+				).name(
+					ObjectValidationRuleSettingConstants.
+						NAME_COMPOSITE_KEY_OBJECT_FIELD_ID
+				).value(
+					String.valueOf(objectField2.getObjectFieldId())
+				).build(),
+				new ObjectValidationRuleSettingBuilder(
+				).name(
+					ObjectValidationRuleSettingConstants.
+						NAME_COMPOSITE_KEY_OBJECT_FIELD_ID
+				).value(
+					String.valueOf(objectField3.getObjectFieldId())
+				).build()));
+
+		Map<String, Serializable> values2 =
+			HashMapBuilder.<String, Serializable>put(
+				"emailAddressRequired", "bob@liferay.com"
+			).put(
+				"listTypeEntryKeyRequired", "listTypeEntryKey1"
+			).put(
+				"numberOfBooksWritten", 5
+			).build();
+
+		ObjectEntry objectEntry = _addObjectEntry(values2);
+
+		_assertCount(1);
+
+		try {
+			_addObjectEntry(values2);
+
+			Assert.fail();
+		}
+		catch (ModelListenerException modelListenerException) {
+			ObjectValidationRuleEngineException
+				objectValidationRuleEngineException =
+					(ObjectValidationRuleEngineException)
+						modelListenerException.getCause();
+
+			List<ObjectValidationRuleResult> objectValidationRuleResults =
+				objectValidationRuleEngineException.
+					getObjectValidationRuleResults();
+
+			Assert.assertEquals(
+				objectValidationRuleResults.toString(), 1,
+				objectValidationRuleResults.size());
+
+			_assertObjectValidationRuleResult(
```

Comment:
@carolmariaabb,
I didn't do that, so as not to increase the complexity of the code, because I would have to change the logic [here](https://github.com/liferay/liferay-portal/blob/6e3df028a104f68300fc5f5b46e0abb6ab36b3eb/modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java#L1344-L1358) .